gmso/core/topology.py

```diff
@@ -1382,9 +1382,21 @@ def iter_sites(self, key, value):
             for site in self._sites:
                 if getattr(site, key) and getattr(site, key).name == value:
                     yield site
-        for site in self._sites:
-            if getattr(site, key) == value:
-                yield site
+        elif isinstance(value, (tuple, list)):
+            containers_dict = {"molecule": Molecule, "residue": Residue}
+            if len(value) == 2:
+                tmp = containers_dict[key](name=value[0], number=value[1])
+            else:
+                tmp = containers_dict[key](
+                    name=value[0], number=value[1], isrigid=value[2]
+                )
```

Comment:
Also, do we need to raise an additional error for trying to use isrigid argument in value for Residues, since we're not making residues allowed to be rigid?

Reply:
Added the code above. Having more check would be good, but it's becoming a bit clumsy. I might need to rethink a better strategy to do these checking (open to suggestion).